experimental/table/server/under/hive/src/main/java/alluxio/table/under/hive/util/PathTranslator.java

```diff
@@ -66,6 +68,10 @@ public PathTranslator addMapping(String alluxioPath, String ufsPath) {
    * @throws IOException if the ufs path is not mounted
    */
   public String toAlluxioPath(String ufsPath) throws IOException {
+    AlluxioURI uri = new AlluxioURI(ufsPath);
```

Comment:
i think this method is called quite often, so it would be more efficient if we did not have to construct AlluxioURI for every ufsPath.  What about adding the mapping to the `PathTranslator` when we update the location of the `partition` and `udbTable`?

Reply:
changed to add the mapping in `HiveTable.updateLocation`